### PR TITLE
Added batch discovery

### PR DESCRIPTION
**Pull request by PrzemyslawTusinski**
Ref https://github.com/openraven/magpie/issues/24

On test and prod credentials there are only jobDefinition defined.
No data for jobQueues and computeEnvironment are present for testing

```
{
  "session" : {
    "id" : "52627d86-fefb-4da9-8301-b0afc3075f22",
    "createdAt" : "2021-03-29T12:43:33.829715Z"
  },
  "pluginPath" : [ "magpie.aws.discovery:batch:jobDefinition" ],
  "contents" : {
    "configuration" : {
      "jobDefinitionName" : "getting-started-job-definition",
      "jobDefinitionArn" : "arn:aws:batch:us-west-2:1234567890:job-definition/getting-started-job-definition:1",
      "revision" : 1,
      "status" : "INACTIVE",
      "type" : "container",
      "parameters" : { },
      "retryStrategy" : null,
      "containerProperties" : {
        "image" : "amazonlinux",
        "vcpus" : 2,
        "memory" : 2000,
        "command" : [ "echo", "hello world" ],
        "jobRoleArn" : null,
        "executionRoleArn" : null,
        "volumes" : [ ],
        "environment" : [ ],
        "mountPoints" : [ ],
        "readonlyRootFilesystem" : null,
        "privileged" : null,
        "ulimits" : [ ],
        "user" : null,
        "instanceType" : null,
        "resourceRequirements" : [ ],
        "linuxParameters" : null,
        "logConfiguration" : null,
        "secrets" : [ ],
        "networkConfiguration" : null,
        "fargatePlatformConfiguration" : null
      },
      "timeout" : null,
      "nodeProperties" : null,
      "tags" : { },
      "propagateTags" : null,
      "platformCapabilities" : null
    },
    "region" : "us-west-2"
  },
  "contentClass" : "com.fasterxml.jackson.databind.node.ObjectNode"
}
```

### CODE REVIEW

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+  private void discoverComputeEnvironments(ObjectMapper mapper, Session session, BatchClient client, Region region, Emitter emitter, Logger logger) {
+    getAwsResponse(
+      () -> client.describeComputeEnvironmentsPaginator().computeEnvironments(),
+      (resp) -> resp.forEach(computeEnvironment -> {
+        var data = mapper.createObjectNode();
+        data.putPOJO("configuration", computeEnvironment.toBuilder());
+        data.put("region", region.toString());
+
+        emitter.emit(new MagpieEnvelope(session, List.of(fullService() + ":computeEnvironment"), data));
```
Comment (PrzemyslawTusinski) [inline magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java:66]: What should be capitalization here?

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  private void discoverJobQueues(ObjectMapper mapper, Session session, BatchClient client, Region region, Emitter emitter, Logger logger) {
+    getAwsResponse(
+      () -> client.describeJobQueuesPaginator().jobQueues(),
+      (resp) -> resp.forEach(computeEnvironment -> {
+        var data = mapper.createObjectNode();
+        data.putPOJO("configuration", computeEnvironment.toBuilder());
+        data.put("region", region.toString());
+
+        emitter.emit(new MagpieEnvelope(session, List.of(fullService() + ":jobQueue"), data));
```
Comment (PrzemyslawTusinski) [inline magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java:80]: What should be capitalization here?

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  private void discoverJobDefinitions(ObjectMapper mapper, Session session, BatchClient client, Region region, Emitter emitter, Logger logger) {
+    getAwsResponse(
+      () -> client.describeJobDefinitionsPaginator().jobDefinitions(),
+      (resp) -> resp.forEach(computeEnvironment -> {
+        var data = mapper.createObjectNode();
+        data.putPOJO("configuration", computeEnvironment.toBuilder());
+        data.put("region", region.toString());
+
+        emitter.emit(new MagpieEnvelope(session, List.of(fullService() + ":jobDefinition"), data));
```
Comment (PrzemyslawTusinski) [inline magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java:95]: What should be capitalization here?

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java (hunk 1 above)
Comment (kickroot) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java:66]: Looks good to me :+1:

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java (hunk 2 above)
Comment (kickroot) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java:80]: Looks good to me :+1:

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java (hunk 3 above)
Comment (kickroot) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BatchDiscovery.java:95]: Looks good to me :+1:

**Review by kickroot — APPROVED** [review]

**Pull request merged**